protocols/dcutr/examples/dcutr.rs

```diff
@@ -155,7 +156,12 @@ fn main() -> Result<(), Box<dyn Error>> {
         dcutr: dcutr::behaviour::Behaviour::new(),
     };
 
-    let mut swarm = SwarmBuilder::new(transport, behaviour, local_peer_id)
+    let executor: Option<Box<dyn Executor + Send>> = match ThreadPool::new() {
+        Ok(tp) => Some(Box::new(tp)),
+        Err(_) => None,
+    };
+
+    let mut swarm = SwarmBuilder::new(transport, behaviour, local_peer_id, executor)
```

Comment:
I'd like this to be more ergonomic, ideally setting an executor here is a one-liner.

Reply:
Yes I'm aware, this is not ideal.

Ideally it would look like this:
```rust
let executor: Option<Box<dyn Executor + Send>> = ThreadPool::new().ok().map(|e| Box::new(e));
```
I'm pretty sure it is equivalent to the code above:
```rust
  let executor: Option<Box<dyn Executor + Send>> = match ThreadPool::new() {
      Ok(tp) => Some(Box::new(tp)),
      Err(_) => None,
  };
```
Correct me I might be wrong here, but it might be one of these classic trait object problems that still bug rust and why I try to avoid them.

Reply:
1. I think `SwarmBuilder` should have two functions: `with_executor` and `without_executor`.
2. We can just change this example to use `AsyncStdExecutor`.

It is not fully equivalent but I don't think there is a reason to show off the `ThreadPoolExecutor` here.

Reply:
Yes, that sounds good. How should I handle `SwarmBuilder::new`?

Reply:
I suggest we deprecate it and have this snippet with the match-statement in there, calling through to either one or the other function.